adi/ad4020.py

```diff
@@ -45,6 +55,38 @@ def sampling_frequency(self, value):
         """Set the sampling frequency."""
         self._set_iio_dev_attr("sampling_frequency", str(value))
 
+    class _channel_adc(attribute):
+        """AD4000 series differential input voltage channel"""
+
+        # AD4000 series ADC channel
+        def __init__(self, ctrl, channel_name, output):
+            self.name = channel_name
+            self._ctrl = ctrl
+            self._output = output
+
+        @property
+        def raw(self):
+            return self._get_iio_attr(self.name, "raw", self._output)
+
+        @property
+        def scale(self):
+            return float(self._get_iio_attr_str(self.name, "scale", self._output))
+
+        @scale.setter
+        def scale(self, value):
+            self._set_iio_attr(self.name, "scale", False, str(Decimal(value).real))
+
+        @property
+        def scale_available(self):
+            """Provides all available scale(gain) settings for the ADC channel"""
```

Comment:
@machschmitt I believe these are all fixed-scale devices. Accessing the scale_available property throws a KeyError.

Reply:
Well yes, at least what is in ADI Linux right now.
Though, the driver will provide a scale_available attribute in the future for configuring span compression.
Some explanation taken from commit log:
AD4000 series of devices have a span compression feature that allows
reducing the ADC input range while keeping the same range of raw output
codes, thus providing a slight increase in measurement precision.
The span compression selection (enable/disable) is done by writing to
the scale attribute. The list of valid scale values is listed by the
scale_available attribute.

@tfcollins, can we have `scale_available` attribute (even if it is not working right now) or would you prefer to only introduce it when the changes get to ADI Linux?

Reply:
@machschmitt noted - although has the decision been made whether the compression will be changeable at run time, or set in the device tree? Typically the decision would be made as the analog front end circuit is being designed, then "set and forget" in software / device tree.
But if it will be run-time adjustable, this makes perfect sense.

Reply:
Hi @mthoren-adi, I was developing this as a runtime feature but it could definitely be set in device tree as you said so that's indeed a good question. Who can I ask about this configuration decision? Part application engineer?